client/getipns.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"context"
+
+	"github.com/ipfs/go-delegated-routing/gen/proto"
+)
+
+func (fp *Client) GetIPNS(ctx context.Context, id []byte) ([][]byte, error) {
+	resps, err := fp.GetIPNSAsync(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	records := [][]byte{}
+	for resp := range resps {
+		if resp.Err != nil {
+			records = append(records, resp.Record)
+		}
+	}
+	if fp.validator != nil {
+		best, err := fp.validator.Select(string(id), records)
+		if err != nil {
+			return nil, err
+		}
+		return [][]byte{records[best]}, nil
+	} else {
+		return records, nil
```

Comment:
If we're removing this if statement then we should probably change GetIPNS to return a single record `[]byte` instead of a set `[][]byte` since we don't have anything interesting to do with the worse/older records.